### DISCUSSION TITLE
Editing glyphs in fontforge

### DISCUSSION THREAD

**Discussion by mil-ad** [ryanoasis/nerd-fonts#1687] [Q&A]
I wanted to learn more about nerdfonts and fontforge by trying to add new glyphs by tweaking current ones. 

I opened the symbols nerd font (i.e. not a patched font) in fontforge and copied a glyph (e.g. from powerline extra set) and pasted it into an unused character. When I generate the TT and use the new glyph its properties (e.g. width, height) doesn't match the original glyph.

Is some metadata lost when I copy paste a glyph in font forge? Is opening/tweaking/exporting Symbols Nerd directly in font forge possible?

**Comment by Finii** [comment]
This seems like questions about Fontforge and not about Nerd Fonts. There are a lot resources/web sites out there that explain Fontforge (I believe ;).

Usually copy-and-paste a glyph in Fontforge does prevent that glyph's specific metadata, but there are font specific metadata that of course are not ... copied over, as changing that would break all preexisting glyphs. Most notably that is the "Em size". The glyph's outline is described in some abstract coordinate system. Usually a "full capital letter" is 1000 or 2048 of this units high, but that can be chosen freely. Sometimes coordinates are integer values, so a bigger value gives higher resolution of places where nodes might be.
If you copy a glyph from one font to another in fontforge and the Em size does not match the new glyph will be smaller/bigger than expected, as the abstract coordinates are used directly and they are not scaled to the new Em size. So usually you should change the Em size of the source font before you copy the glyph to match the target font's Em size. That will scale all glyphs abstract coordinates (if you have the "Scale Outlines" option ticked).

![image](https://github.com/user-attachments/assets/8cf9dd0d-cb88-4963-95fe-387a8fe1c0d2)

> Is opening/tweaking/exporting Symbols Nerd directly in font forge possible?

Shure. Exporting (called generating in Fontforge) can be a bit tricky because there are so many options.
For the SymbolsOnly font as `ttf` choose `TrueType` and not one of the TrueType variants:

![image](https://github.com/user-attachments/assets/b7cd9434-4d8b-4994-a474-effe06be0ffe)

and maybe these options are ok for that usecase:

![image](https://github.com/user-attachments/assets/44e6dda3-3274-4715-a1a8-456ee019a943)

There are no hints in the source so the TrueType hints option does not matter, but generally one wants the hints to be preserved.

If you want to create a `.otf` font use `OpenType (CFF)`.

Both `ttf` and `otf` are OpenType fonts nowadays, they just a different model for the curve description. OpenType CFF uses Postscript's description while TrueType uses the simpler TrueType description. Usually it is best to not change that for letters (i.e. keep as the sourcefont) - esp because the hints will break if changed from ttf to otf and vice versa.

Further information about fonts in general, can be found for example at
https://learn.microsoft.com/en-us/typography/opentype/

And Yannis Haralambous' book is still relevant I guess (at least I still use it and know of no better one)

![image](https://github.com/user-attachments/assets/ed5c68ee-1c29-46c5-9718-6d96708c9cc9)
